### ISSUE TITLE
AlmaLinux 8&9 container scan not detecting OS properly

### ISSUE THREAD

**Issue by tas50** [mondoohq/cnspec#164]
**Describe the bug**
AlmaLinux 8 container scan not detecting OS which breaks service, package, and basically the whole mondoo scan.

**To Reproduce**
1. `cnspec scan container almalinux:8`
2. See errors

**Expected behavior**
Scans w/o error.

**Screenshots**
```
 ~/dev  cnspec scan container almalinux:8
→ loaded configuration from /Users/tsmith/.config/mondoo/mondoo.yml using source default
→ using service account credentials
→ discover related assets for 1 asset(s)
→ resolved assets resolved-assets=1
→ syncing assets
→ connecting to asset index.docker.io/library/almalinux@56ec0a505cbd (container-image)
! resolver.db> failed to store data, types don't match asset=//assets.api.mondoo.app/spaces/practical-visvesvaraya-957532/assets/2H3sJ4Pw5FDqr8SF5IvLQLxHU7K checksum=aqYNhAltcEb8Us9rrSndGyONs5+/eu8zKLjtc3MhT1V9wzIHoZniQylGNL6k9+ftj8MnoMI/18dd3mMzIxBBhA== data={"type":"\u001bservice"} expected=block received=service
x failed to send datapoints error="1 error occurred:\n\t* failed to store data for \"aqYNhAltcEb8Us9rrSndGyONs5+/eu8zKLjtc3MhT1V9wzIHoZniQylGNL6k9+ftj8MnoMI/18dd3mMzIxBBhA==\", types don't match: expected block, got service\n\n"


Asset: index.docker.io/library/almalinux@56ec0a505cbd
=====================================================
Data queries:
packages.list.length: 149
error: Query encountered errors:
failed to create resource 'service': could not retrieve service list
service: no data available
error: Query encountered errors:
could not determine kernel version
kernel.installed: []
error: Query encountered errors:
could not parse uptime:
os.uptime: 3 seconds
platform.arch: "amd64"
error: Query encountered errors:
run command not supported on this transport
command.stderr: ""
error: Query encountered errors:
run command not supported on this transport
if: no data available
platform.eol.date: 2029-05-31 17:00:00 -0700 PDT
error: Query encountered errors:
could not detect hostname
os.hostname: ""
platform.vulnerabilityReport: {
  platform: {
    arch: "amd64"
    name: "almalinux"
    release: "8.6"
    title: "AlmaLinux 8.6 (Sky Tiger), Docker Image"
  }
  published: "2022-11-04T01:34:45Z"
  stats: {
    advisories: {}
... 7 more lines ...
mondoo.jobEnvironment: {
  id: "client.mondoo.com"
  name: "Mondoo Client"
}
platform.release: "8.6"
mondoo.version: "unstable"
platform.title: "AlmaLinux 8.6 (Sky Tiger), Docker Image"

Controls:
! Error: Ensure events that modify date and time information are collected
✕ Fail:  Ensure audit logs are not automatically deleted
! Error: Ensure tftp server is stopped and not enabled
! Error: Ensure file deletion events by users are collected
✓ Pass:  Ensure secure permissions on /etc/group are set
! Error: Ensure rsyslog Service is enabled
! Error: Ensure HTTP servers are stopped and not enabled
✓ Pass:  Ensure X Window System is not installed
✕ Fail:  F   0  Ensure IPv6 router advertisements are not accepted
! Error: Ensure Avahi server is stopped and not enabled
! Error: Ensure core dumps are restricted
✓ Pass:  Ensure default group for the root account is GID 0
! Error: Ensure rsh server is stopped and not enabled
! Error: Ensure NFS and RPC are stopped and not enabled
✓ Pass:  Ensure system accounts are non-login
✓ Pass:  Ensure UID_MIN is set to 1000
✓ Pass:  Ensure no duplicate group names exist
! Error: Ensure IMAP and POP3 server is stopped and not enabled
✕ Fail:  F   0  Ensure source routed packets are not accepted
✕ Fail:  C  40  Ensure rsyslog default file permissions configured
! Error: Ensure events that modify the system's network environment are collected
! Error: Ensure session initiation information is collected
✓ Pass:  Ensure root group is empty
. Skipped: Ensure secure permissions on all log files are set
✕ Fail:  C  40  Ensure Advanced Intrusion Detection Environment (AIDE) is installed
✕ Fail:  C  40  Ensure IP forwarding is disabled
✕ Fail:  F   0  Ensure broadcast ICMP requests are ignored
✕ Fail:  C  40  Ensure journald is configured to send logs to rsyslog
✓ Pass:  Ensure secure permissions on /etc/gshadow are set
✕ Fail:  F   0  Ensure secure ICMP redirects are not accepted
! Error: Ensure FTP server is stopped and not enabled
✕ Fail:  F   0  Ensure packet redirect sending is disabled
✕ Fail:  C  40  Ensure rsyslog is installed
✕ Fail:  F   0  Ensure access to the su command is restricted
✓ Pass:  Ensure no duplicate GIDs exist
! Error: Ensure rsync service is stopped and not enabled
! Error: Ensure CUPS is stopped and not enabled
✓ Pass:  Ensure secure permissions on /etc/shadow are set
! Error: Ensure filesystem integrity is regularly checked
✕ Fail:  D  20  Ensure audit log storage size is configured
! Error: Ensure talk server is stopped and not enabled
! Error: Ensure LDAP server is stopped and not enabled
✕ Fail:  F   0  Ensure Reverse Path Filtering is enabled
! Error: Ensure login and logout events are collected
✕ Fail:  B  60  Ensure journald is configured to compress large log files
! Error: Ensure events that modify the system's Mandatory Access Controls are collected
✕ Fail:  F   0  Ensure TCP SYN Cookies is enabled
✕ Fail:  C  40  Ensure journald is configured to write logfiles to persistent disk
✓ Pass:  Ensure secure permissions on /etc/gshadow- are set
✓ Pass:  Ensure no known platform CVEs exist
✓ Pass:  Ensure the platform is not End-of-Life
✓ Pass:  Ensure no known platform advisories exist
! Error: Ensure events that modify user/group information are collected
✕ Fail:  D  20  Ensure suspicious packets are logged
✓ Pass:  Ensure each user is a member of a group
! Error: Ensure SNMP server is stopped and not enabled
✓ Pass:  Ensure secure permissions on /etc/passwd are set
! Error: Ensure NIS server is stopped and not enabled
✓ Pass:  Platform is not end-of-life
! Error: Ensure Samba is stopped and not enabled
! Error: Ensure system administrator actions (sudolog) are collected
! Error: Ensure discretionary access control permission modification events are collected
! Error: Ensure the audit configuration is immutable
! Error: Ensure mail transfer agent is configured for local-only mode
. Skipped: Ensure auditing for processes that start prior to auditd is enabled
✓ Pass:  Ensure all GIDs in /etc/passwd exist in /etc/group
. Skipped: Ensure shadow group is empty
! Error: Ensure HTTP Proxy server is stopped and not enabled
✕ Fail:  F   0  Ensure secure permissions on /etc/group- are set
✕ Fail:  C  40  Ensure system is disabled when audit logs are full
! Error: Ensure telnet server is stopped and not enabled
! Error: Ensure unsuccessful unauthorized file access attempts are collected
✕ Fail:  F   0  Ensure address space layout randomization (ASLR) is enabled
✓ Pass:  Ensure no duplicate UIDs exist
! Error: Ensure successful file system mounts are collected
✓ Pass:  Ensure secure permissions on /etc/shadow- are set
✓ Pass:  Ensure no duplicate user names exist
! Error: Ensure kernel module loading and unloading is collected
✕ Fail:  F   0  Ensure ICMP redirects are not accepted
! Error: Ensure sudo logging is enabled
✕ Fail:  F   0  Ensure bogus ICMP responses are ignored
✕ Fail:  D  20  Ensure auditd is installed
! Error: Ensure changes to system administration scope (sudoers) is collected
! Error: Ensure DNS server is stopped and not enabled
✕ Fail:  F   0  Ensure secure permissions on /etc/passwd- are set
! Error: Ensure DHCP server is stopped and not enabled
✓ Pass:  Ensure prelink is disabled
! Error: Ensure auditd service is enabled

Vulnerabilities:
■ No advisories found (passed)

Overall CVSS score: 0.0


Summary (1 assets)
==================

Target:     index.docker.io/library/almalinux@56ec0a505cbd
Score:      B    66/100     (100% completed)
✓ Passed:   ████ 25% (22)
✕ Failed:   ████ 28% (25)
! Errors:   ██████ 43% (38)
» Skipped:  █ 3% (3)

Policies:
F   0  Linux Security by Mondoo
.  ..  Platform Overview Information by Mondoo
A 100  Platform Vulnerability Policy by Mondoo
A 100  Platform End-of-Life Policy by Mondoo

Report URL: FOO

To get more information, please run this scan with "-o full".
```

<img width="1179" alt="image" src="https://user-images.githubusercontent.com/1015200/199866659-5515267a-3979-4312-bd80-c25db5699da4.png">

**Comment by chris-rock** [comment]
@tas50 the output is not matching the title, packages are discovered, vulnerability scan worked. The only thing I see here is that the `Platform Overview Information by Mondoo` query pack is not handing containers well in general which is not related to almalinux 8 or 9. Can you confirm that?

**Comment by tas50** [comment]
@chris-rock Yes packages do actually appear to be correct on this. Just failures for all the service checks, but that may just be the container image scan.

![image](https://user-images.githubusercontent.com/1015200/200039807-d301ca48-3c24-416e-bee3-cf7f891bf5e3.png)

**Comment by tas50** [comment]
Resolved as of 7.5
